velox/type/Variant.cpp

```diff
@@ -227,11 +237,13 @@ std::string variant::toJson(const TypePtr& type) const {
       std::string b{};
       b += "[";
       bool first = true;
+      uint32_t idx = 0;
+      auto arrayElementType = type->childAt(0);
```

Comment:
The array can be empty. We should handle that.

Reply:
If the array is empty, then the return will be `[]`. We have a test for it. Isn't that we expect? Should we check `type->size()` to 0. Not sure if we would get that.

Reply:
From the code, it looks like `auto arrayType = ARRAY();` is allowed. Can you check?

Reply:
No, this is not allowed. We have only one definition and its expects atleast one arg.
```
std::shared_ptr<const ArrayType> ARRAY(TypePtr elementType) {
  return std::make_shared<const ArrayType>(std::move(elementType));
}
```

Reply:
elementType can be a `nullptr`. But `type->childAt(0)` will always work.